app/src/main/java/org/fossasia/susi/ai/Signup/SignUpActivity.kt

```diff
@@ -0,0 +1,202 @@
+        val alert = alertDialog.create()
+        alert.show()
+        val yes = alert.getButton(DialogInterface.BUTTON_POSITIVE)
+        val no = alert.getButton(DialogInterface.BUTTON_NEGATIVE)
+        yes.setTextColor(resources.getColor(R.color.md_blue_500))
+        no.setTextColor(resources.getColor(R.color.md_red_500))
+    }
+
+
+    override fun alertSuccess() {
+        val dialogClickListener = DialogInterface.OnClickListener { dialogInterface, i -> finish() }
+        val alertTitle = resources.getString(R.string.signup)
+        val alertMessage = resources.getString(R.string.signup_msg)
+        val successAlertboxHelper = AlertboxHelper(this@SignUpActivity, alertTitle, alertMessage, dialogClickListener, null, resources.getString(R.string.ok), null, resources.getColor(R.color.md_blue_500))
+        successAlertboxHelper.showAlertBox()    }
+
+    override fun alertFailure() {
+        val dialogClickListener = DialogInterface.OnClickListener { dialogInterface, i ->
+            val intent = Intent(this@SignUpActivity, LoginActivity::class.java)
+            intent.flags = Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_NEW_TASK
+            startActivity(intent)
+            finish()
+        }
+        val dialogClickListenern = DialogInterface.OnClickListener { dialogInterface, i ->
+            val intent = Intent(this@SignUpActivity, ForgotPasswordActivity::class.java)
+            startActivity(intent)
+            finish()
+        }
+        val alertTitle = resources.getString(R.string.error_email)
+        val alertMessage = resources.getString(R.string.error_msg)
+        val failureAlertboxHelper = AlertboxHelper(this@SignUpActivity, alertTitle, alertMessage, dialogClickListener, dialogClickListenern, resources.getString(R.string.ok), resources.getString(R.string.forgot_pass_activity), resources.getColor(R.color.md_blue_500))
+        failureAlertboxHelper.showAlertBox()
+    }
+
+    override fun alertError(title: String, message: String) {
+        val errorAlertboxHelper = AlertboxHelper(this@SignUpActivity, title, message, null, null, resources.getString(R.string.ok), null, Color.BLUE)
+        errorAlertboxHelper.showAlertBox()
+    }
+
+    override fun setErrorEmail(msg: String) {
+        email?.setError(msg)
+    }
+
+    override fun setErrorPass(msg: String) {
+        password?.setError(msg)
+    }
+
+    override fun setErrorConpass(msg: String) {
+        confirm_password?.setError(msg)
+    }
+
+    override fun setErrorUrl(msg: String) {
+        input_url?.setError(msg)
+    }
+
+    override fun enableSignUp(bool: Boolean) {
+        sign_up?.setEnabled(bool)
+    }
+
+    override fun isPersonalServer(): Boolean? {
+        return personal_server?.isChecked()
+    }
+
+    override fun clearField() {
+        CredentialHelper.clearFields(email, password, confirm_password)
+    }
+
+    override fun checkIfEmptyUrl(): Boolean {
```

Comment:
Shouldn't this go in interactor. Only add code in SignUpActivity which updates view. All business logic should go in Interactor.